### PR TITLE
Frontend changes for user auth + user context (storing the user)

### LINKED ISSUE #90: User Registration functionality
## Feature Description
Code for user registration, which involves creating a JWT token on sign up and returning it with the new user object to the front end

blocked by: #82

## Problem You're Trying to Solve
Users must be able to register for the website in order to use it

## Related External Resources
[section 4.4.1](https://docs.google.com/document/d/1DNqexMBf4MpYPCxWsT3plUDAjDSbdJwUzw-UyZX-vDs/edit#heading=h.vbv6udt3y8om)

### PR DESCRIPTION
**Pull request by Scott-Kenning**
# Description
Frontend changes for user auth, + storing user in context

Closes #90

## How to Test
Run, create account, login, navigate to profile page

## Checklist
- [x] The code includes tests if relevant
- [x] I have *actually* self-reviewed my changes and done QA

### CODE REVIEW

**Review by CaelumD25 — APPROVED** [review]
It looks okay to me, but the tests do need to be passing before merging. As long as that works I'm happy with this

File: apps/frontend/src/_axios_instance.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import axios from "axios";
 
 // Create an instance of axios with custom default configurations
 const _axios_instance = axios.create({
-  baseURL: "https://whateverthiswillbe.com/api/", // Your custom base URL
+  baseURL: "http://localhost/api/", // Your custom base URL
```
Comment (MNThomson) [inline apps/frontend/src/_axios_instance.tsx:5]: This should be whatever hostname is in the url bar

File: apps/frontend/src/contexts/UserProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  const setUserAndStopLoading = (user: User) => {
+    localStorage.setItem("user", JSON.stringify(user));
+    setUser(user);
+    setLoading(false);
+  };
+
+  const logout = async () => {
+    setUser(null);
+    localStorage.removeItem("user");
+    Cookies.remove("token");
```
Comment (MNThomson) [inline apps/frontend/src/contexts/UserProvider.tsx:28]: Front-end will not have any access to the token. It's not being sent in the http response body. It's in a `Set-Cookie` header that has `httponly = true` meaning that the front-end CANNOT read (not that you shouldn't, but that the browser will not give it to you)

**Review by MNThomson — CHANGES_REQUESTED** [review]
Not sure why there is anything related to cookies in this pr. The front-end physically (I guess virtually???) cannot access the auth cookie/token

**Review by MNThomson — APPROVED** [review]

**Review by islandj123 — APPROVED** [review]
tis code

**Pull request merged**